scripts/deptool.py

```diff
@@ -0,0 +1,728 @@
+                            header_row, column_widths
+                        )
+                    ]
+                )
+                for row_name, row in zip(row_names, nested_dict)
+            ]
+        )
+        + row_right_separator
+    )
+
+    return table_string
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="CFEngine dependencies enumeration tool"
+    )
+    parser.add_argument(
+        "branches",
+        nargs="*",
+        help="List of branches to process",
+        default=ACTIVE_BRANCHES,
+    )
+    parser.add_argument(
+        "--to-cdx-sbom",
+        help="Output to a CycloneDX SBOM file with an optionally specified path",
+        nargs="?",
+        const="sbom.cdx.json",
+        default=None,
+        dest="cdx_sbom_path",
+    )
+    parser.add_argument(
+        "--to-json",
+        help="Output to a JSON file with an optionally specified path",
+        nargs="?",
+        const="deps.json",
+        default=None,
+        dest="json_path",
+    )
+    parser.add_argument(
+        "--compare",
+        action="store_true",
+        help="Compare branches in pairs instead of processing each branch individually for the displayed Markdown table",
+    )
+    parser.add_argument(
+        "--skip-unchanged",
+        action="store_true",
+        help="Skip dependencies with all versions identical when using --compare",
+    )
+    parser.add_argument(
+        "--patch",
+        action="store_true",
+        help="Modify the README with updated dependency tables in addition to displaying them",
+    )
+    parser.add_argument(
+        "--no-info",
+        action="store_true",
+        help="Disable informational messages",
+    )
+    parser.add_argument(
+        "--root",
+        help="Optionally specify buildscripts root directory path",
+        dest="repo_path",
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    if args.compare and len(args.branches) % 2 == 1:
+        log.warning("comparing with an odd number of versions")
+
+    dr = DepsReader(repo_path=args.repo_path, log_info=not args.no_info)
+
+    if args.cdx_sbom_path:
+        dr.write_cdx_sbom(args.cdx_sbom_path, args.branches)
+
+    if args.json_path:
+        dr.write_deps_json(args.json_path, args.branches)
+
+    if args.patch or not args.compare:
+        updated_readme, updated_agent_table, updated_hub_table = (
+            dr.updated_deps_markdown_table(args.branches)
+        )
+
+    if args.compare:
+        comparison_table = dr.comparison_md_table(args.branches, args.skip_unchanged)
+        print(comparison_table)
+    else:
+        print("### Agent Dependencies:\n")
+        print(updated_agent_table)
+        print("\n### Enterprise Hub dependencies:\n")
+        print(updated_hub_table)
+
+    if args.patch:
+        dr.patch_readme(updated_readme)
+
+
+if __name__ == "__main__":
+    ret = main()
+    sys.exit(ret)
```

Comment:
This technically creates a global variable which could be avoided;

```suggestion
if __name__ == "__main__":
    sys.exit(main())
```

Reply:
(but this is not a big deal, of course)